File: Data Structure and Algorithms I Lab ( B )/code/sample solution/Problem 2.c
#include <stdio.h>
#include <stdlib.h>

int inversionCount(int Arr[], int n)
{
    int i, j;
    int inversion_count = 0;       //the counting will begin 0

    for(i = 0; i < n-1; i++){
        for(j = i + 1; j < n; j++)
        {
            if(Arr[i] > Arr[j])
            {
                inversion_count++;
            }
        }
    }
    return inversion_count;
}

void printArray(int Arr[], int size)
{
	for (int k = 0; k < size; k++)
    printf("%d ", Arr[k]);
	printf("\n");
}

int main()
{

    int Arr[20];
    int n = sizeof(Arr) / sizeof(Arr[0]);
    for(int i = 0; i < 20; i++)   //random array input
      {
          Arr[i] = rand()%500;
      }

    printf("Unsorted Array: \n");
    printArray(Arr, n);

    printf("The inversion count of Arr is %d \n", inversionCount(Arr, n));
    

    return 0;
}

File: Data Structure and Algorithms I Lab ( B )/code/java/src/com/company/Problem2.java
package com.company;

/* Problem-02 */

import static com.company.Inversion.A;


class Inversion{
    static int[] A=new int[] {1, 9, 6, 4, 5};
    static int Count(int c){
        int inversion_count=0;
        for(int i=0; i<c-1; i++){
            for(int j=i+1; j<c; j++){
                if(A[i]>A[j]){
                    inversion_count++;
                }
            }
        }
        return inversion_count;
    }
}
public class Problem2 {
    public static void main(String[] args) {

        System.out.println("The inversion count of A is : "+ Inversion.Count(A.length));
    }
}
